src/ucs/sys/sock.c

```diff
@@ -366,14 +370,33 @@ ucs_status_t ucs_socket_server_init(const struct sockaddr *saddr, socklen_t sock
         goto err_close_socket;
     }
 
-    status = ucs_sockaddr_get_port(saddr, &port);
-    if (status != UCS_OK) {
-        goto err_close_socket;
+    if ((ports_range_start != 0) && (ports_range_start == ports_range_end)) {
+        /* non zero single value port */
+        retry = 0;
+    } else {
+        /* port value is zero or a range of ports is given */
+        retry = 1;
     }
 
+    rand_seed = ucs_generate_uuid((uintptr_t)saddr);
+
     do {
+        if ((ports_range_start != 0) || (ports_range_end != 0)) {
+            port = (uint16_t)((rand_r(&rand_seed) %
+                              (ports_range_end -
+                               ports_range_start + 1)) +
+                              ports_range_start);
```

Comment:
can you pls add ucs_rand() and ucs_rand_range(min, max) functions to wrap this (with a global seed)

Reply:
what do you mean by 'global seed' ?

Reply:
global variable in use to be the rand start which initialized with rand seed from get_uuid

Reply:
so add a param to ucp context's config?

Reply:
no, it should be global variable inside UCS, unrelated to UCP